armi/operators/operator.py

```diff
@@ -355,10 +355,13 @@ def _cycleLoop(self, cycle, startingCycle):
         if halt:
             return False
 
+        # read total core power from settings (power or powerDensity)
+        basicPower = self.cs["power"]
+        if basicPower == 0 and self.cs["powerDensity"] > 0:
+            basicPower = self.cs["powerDensity"] * self.r.core.getHmm()
+
         for timeNode in range(startingNode, int(self.burnSteps[cycle])):
-            self.r.core.p.power = (
-                self.powerFractions[cycle][timeNode] * self.cs["power"]
-            )
+            self.r.core.p.power = self.powerFractions[cycle][timeNode] * basicPower
```

Comment:
Huh, this ends up being much easier than I expected. I thought we were going to have to replace calls to `cs["power"]` all over the place. But because people are already used to the fact that power can change via `powerFractions`, I guess they are used to always going to the operator to get the current power level.

I did find (at least) one instance of a direct call to `cs["power"]` in an internal implementation of a custom operator class. So we'll have to fix that as well.